kpet/schema.py

```diff
@@ -396,13 +396,20 @@ class List(Node):
     """
     List schema, with every element matching a single specified schema.
     """
-    def __init__(self, element_schema):
+    def __init__(self, element_schema, min_len=None):
         assert isinstance(element_schema, Node)
         super().__init__(list)
         self.element_schema = element_schema
+        self.min_len = min_len
 
     def validate(self, data):
         super().validate(data)
+
+        if self.min_len and len(data) < self.min_len:
+            raise Invalid(
+                "This list must have at least {} elements!", len(data)
```

Comment:
It might be difficult to understand what the message points at in an error output with this message, so please just write something like `"List must be at least {} elements long, but only has {}"`.